src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -27,44 +27,39 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     public sealed partial class AnalyzerFileReference : AnalyzerReference, IEquatable<AnalyzerReference>
     {
         private readonly string _fullPath;
-        private readonly Func<string, Assembly> _getAssembly;
+        private readonly Assembly _assembly;
+        private readonly string _displayName;
 
-        private string _lazyDisplayName;
         private ImmutableArray<DiagnosticAnalyzer> _lazyAllAnalyzers;
         private ImmutableDictionary<string, ImmutableArray<DiagnosticAnalyzer>> _lazyAnalyzersPerLanguage;
-        private Assembly _lazyAssembly;
         private static string s_diagnosticNamespaceName = string.Format("{0}.{1}.{2}", nameof(Microsoft), nameof(CodeAnalysis), nameof(Diagnostics));
         private ImmutableDictionary<string, ImmutableHashSet<string>> _lazyAnalyzerTypeNameMap;
 
         public event EventHandler<AnalyzerLoadFailureEventArgs> AnalyzerLoadFailed;
 
         /// <summary>
-        /// Creates an AnalyzerFileReference with the given <paramref name="fullPath"/>.
+        /// Creates an <see cref="AnalyzerFileReference"/> with the given <paramref name="fullPath"/> and <paramref name="assembly"/>.
         /// </summary>
         /// <param name="fullPath">Full path of the analyzer assembly.</param>
-        /// <param name="getAssembly">Function for loading the analyzer assembly</param>
-        public AnalyzerFileReference(string fullPath, Func<string, Assembly> getAssembly)
+        /// <param name="assembly">The analyzer assembly</param>
+        public AnalyzerFileReference(string fullPath, Assembly assembly)
```

Comment:
Additionally, with the new design that we need to specify all analyzer dependencies as an analyzer, we might unnecessarily load assemblies which have no analyzers upfront.
Can we instead change the constructor to take `IAnalyzerAssemblyLoader` as the second parameter?